modules/basics/src/main/java/com/opengamma/strata/basics/currency/Money.java

```diff
@@ -0,0 +1,223 @@
+    return new Money(currency, roundedAmount);
+  }
+
+  /**
+   * Obtains an instance of {@code Money} for the specified currency and amount.
+   *
+   * @param currency the currency the amount is in
+   * @param amount the amount of the currency to represent
+   * @return the currency amount
+   */
+  public static Money of(Currency currency, double amount) {
+    BigDecimal roundedAmount = new BigDecimal(amount)
+        .setScale(currency.getMinorUnitDigits(), BigDecimal.ROUND_HALF_EVEN);
+    return new Money(currency, roundedAmount);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Creates an instance.
+   *
+   * @param currency the currency
+   * @param amount the amount
+   */
+  private Money(Currency currency, BigDecimal amount) {
+    this.currency = currency;
+    this.amount = amount;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the currency.
+   * <p>
+   * For example, in the value 'GBP 12.34' the currency is 'GBP'.
+   *
+   * @return the currency
+   */
+  public Currency getCurrency() {
+    return currency;
+  }
+
+  /**
+   * Gets the amount of the currency as an instance of {@link BigDecimal}.
+   * The amount will be rounded to the currency specifications.
+   * <p>
+   * For example, in the value 'GBP 12.34' the amount is 12.34.
+   *
+   * @return the amount
+   */
+  public BigDecimal getAmount() {
+    return amount;
+  }
+
+  /**
+   * Gets the amount of the currency as a double primitive.
+   * The amount will be rounded to the currency specifications.
+   * <p>
+   * For example, in the value 'GBP 12.34' the amount is 12.34.
+   *
+   * @return the amount
+   */
+  public double getAmountAsDouble() {
+    return amount.doubleValue();
+  }
+
+
+  //-------------------------------------------------------------------------
+  /**
+   * Converts this amount to an equivalent amount the specified currency.
+   * <p>
+   * The result will be expressed in terms of the given currency, converting
+   * using the specified FX rate.
+   * <p>
+   * For example, if this represents 'GBP 100' and this method is called with
+   * arguments {@code (USD, 1.6)} then the result will be 'USD 160'.
+   *
+   * @param resultCurrency the currency of the result
+   * @param fxRate the FX rate from this currency to the result currency
+   * @return the converted instance, which should be expressed in the specified currency
+   * @throws IllegalArgumentException if the FX is not 1 when no conversion is required
+   */
+  public Money convertedTo(Currency resultCurrency, double fxRate) {
+    if (currency.equals(resultCurrency)) {
+      if (DoubleMath.fuzzyEquals(fxRate, 1d, 1e-8)) {
+        return this;
+      }
+      throw new IllegalArgumentException("FX rate must be 1 when no conversion required");
+    }
+    return Money.of(resultCurrency, amount.multiply(new BigDecimal(fxRate)).doubleValue());
```

Comment:
This does an unnecessary conversion from `BigDecimal` to `double` and then back to `BigDecimal`.